packages/dd-trace/src/debugger/devtools_client/snapshot/processor.js

```diff
@@ -209,8 +216,10 @@ function toMap (type, pairs, maxLength) {
   return result
 }
 
-function toSet (type, values, maxLength) {
+function toSet (type, values, maxLength, timeBudgetReached) {
+  if (timeBudgetReached === true) return notCapturedTimeBudget(type)
   if (values === undefined) return notCapturedDepth(type)
+  if (values.every(({ value }) => Object.hasOwn(value, timeBudgetSym))) return notCapturedTimeBudget(type)
```

Comment:
This is actually handled different for arrays and objects. Is that intentional? I would have expected to work the same in all of these cases.